### PR TITLE
[imagemagick] update versions, add install checks

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

File: roles/imagemagick/tasks/imagemagick.yml [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
-
-- name: download release
-  get_url: url=http://www.imagemagick.org/download/releases/{{ im_zip }} dest={{ magick_path }}/{{ im_zip }}
-
-- name: unzip tarball
-  unarchive: src={{ magick_path }}/{{ im_zip }} dest={{ magick_path }} copy=no
+# ImageMagick regularly removes old tarballs, so use a mirror
+- name: download ImageMagick
+  get_url:
+    url: https://dl.bintray.com/homebrew/mirror/ImageMagick-6.9.3-6.tar.xz
```
Comment (dunn) [inline roles/imagemagick/tasks/imagemagick.yml]: Forgot to use the version variable here, will update.

File: roles/imagemagick/tasks/im_libraries.yml [hunk 2]
```diff
@@ -6,21 +6,21 @@
   become: yes
   yum: name={{ item }} state=present
   with_items:
```
Comment (acozine) [inline roles/imagemagick/tasks/im_libraries.yml:8]: ++ for alphabetizing the library names

File: roles/imagemagick/tasks/main.yml [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
-#   EG:
-#     - { role: imagemagick, imagemagick_ver: '6.6' }   # will install the highest version matching the 6.6 prefix
+# Installs the most recent release of ImageMagick, if you need a
+# specific major or minor release, set the `imagemagick_ver:` variable
 
 - include: im_libraries.yml
-- include: openjpg.yml
+
+- stat: path=/usr/local/lib/libtiff.a
+  register: tiff_stat
```
Comment (acozine) [inline roles/imagemagick/tasks/main.yml:12]: Won't these checks allow different versions of libtiff etc. to pass unchallenged? Given the issues we've seen in the past with outdated versions, I'm hesitant to do this.

Comment (dunn) [inline roles/imagemagick/tasks/main.yml:12]: Oh, good point, I'll stat the versioned libs.

Comment (dunn) [inline roles/imagemagick/tasks/main.yml:12]: Should I use the full version—in libtiff's case `5.2.4`, or just the major version (5)?

Comment (acozine) [inline roles/imagemagick/tasks/main.yml:12]: I think the full version (which you've done) is a good idea.

File: roles/imagemagick/defaults/main.yml [hunk 4]
```diff
@@ -3,7 +3,17 @@
...
 magick_path: "{{ install_path }}/imagemagick_sources" # needs double-quotes for yaml syntax
+
 openjpg_ver: '2.1.0'
-libtiff_ver: '4.0.5'
-libpng_ver: '1.6.20'
-imagemagick_ver: '.' # call the role with imagemagick_ver: '6.8' to get the most recent 6.8 release, '.' to match the newest release
+openjpg_256: 1232bb814fd88d8ed314c94f0bfebb03de8559583a33abbe8c64ef3fc0a8ff03
+
+libtiff_ver: '4.0.6'
+libtiff_lib_ver: 5.2.4
```
Comment (acozine) [inline roles/imagemagick/defaults/main.yml:11]: needs single quotes?

Comment (dunn) [inline roles/imagemagick/defaults/main.yml:11]: I don't think so, the version numbers are unquoted in the ffmpeg PR and it seems to work.

Comment (acozine) [inline roles/imagemagick/defaults/main.yml:11]: okay, can you take them off the others then?

Comment (dunn) [inline roles/imagemagick/defaults/main.yml:11]: OK!

**Comment by acozine** [comment]
Tested in Vagrant with a full build.

**Pull request merged**